src/wallet/wallettool.cpp

```diff
@@ -168,7 +168,8 @@ bool ExecuteWalletToolFunc(const ArgsManager& args, const std::string& command)
         DatabaseOptions options;
         ReadDatabaseArgs(args, options);
         options.require_existing = true;
-        const std::shared_ptr<CWallet> wallet_instance = MakeWallet(name, path, args, options);
+        // NOTE: We need to skip initialisation of the m_used flag, or else the address book count might be wrong
```

Comment:
3ee478b9cba305dc87de1ad093ca14a89e58b516 a test demonstrating the bug would be nice

Reply:
This is already caught by an existing test (`test/functional/tool_wallet.py` IIRC)